### PR TITLE
Create creating_synthetic_tokens.md

### PR DESCRIPTION
**Pull request by rcai1**
Tutorial for creating synthetic tokens locally

### CODE REVIEW

File: documentation/tutorials/creating_synthetic_tokens.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
+This tutorial will show you how to create synthetic tokens from the command line using UMA’s synthetic token template. Before beginning this tutorial, please make sure your environment is set up correctly by following the instructions in the “Prerequisites” section. After completing this section, your environment will:
```
Comment (chrismaree) [inline documentation/tutorials/creating_synthetic_tokens.md]: This should start with a big heading to be consistent with other tutorials.
```suggestion
# Creating Synthetic Tokens from Truffle console

This tutorial will show you how to create synthetic tokens from the command line using UMA’s synthetic token template. Before beginning this tutorial, please make sure your environment is set up correctly by following the instructions in the “Prerequisites” section. After completing this section, your environment will: 
```

File: documentation/tutorials/creating_synthetic_tokens.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+ 
+3. Create an instance of the expiring multiparty creator (the contract factory for synthetic tokens). This command should return “undefined”. 
+```
+const empCreator = await ExpiringMultiPartyCreator.deployed()
+```
+ 
+4. Define the parameters for the synthetic tokens you would like to create. For documentation surrounding the meanings of each parameter, please see this page. 
+ 
+```
+const constructorParams = { expirationTimestamp: "123456789000", withdrawalLiveness: "1000", siphonDelay: "100000", collateralAddress: TestnetERC20.address, tokenFactoryAddress: TokenFactory.address, priceFeedIdentifier: web3.utils.utf8ToHex("UMATEST"), syntheticName: "Test UMA Token", syntheticSymbol: "UMATEST", liquidationLiveness: "1000", collateralRequirement: { rawValue: web3.utils.toWei("1.5") }, disputeBondPct: { rawValue: web3.utils.toWei("0.1") }, sponsorDisputeRewardPct: { rawValue: web3.utils.toWei("0.1") }, disputerDisputeRewardPct: { rawValue: web3.utils.toWei("0.1") } };
```
Comment (chrismaree) [inline documentation/tutorials/creating_synthetic_tokens.md]: I think this should not have the `;` puncuation market.
```suggestion
const constructorParams = { expirationTimestamp: "123456789000", withdrawalLiveness: "1000", siphonDelay: "100000", collateralAddress: TestnetERC20.address, tokenFactoryAddress: TokenFactory.address, priceFeedIdentifier: web3.utils.utf8ToHex("UMATEST"), syntheticName: "Test UMA Token", syntheticSymbol: "UMATEST", liquidationLiveness: "1000", collateralRequirement: { rawValue: web3.utils.toWei("1.5") }, disputeBondPct: { rawValue: web3.utils.toWei("0.1") }, sponsorDisputeRewardPct: { rawValue: web3.utils.toWei("0.1") }, disputerDisputeRewardPct: { rawValue: web3.utils.toWei("0.1") } }
```

File: documentation/tutorials/creating_synthetic_tokens.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+```
+ 
+5. Before the contract for the synthetic tokens can be created, the price identifier for the synthetic tokens must be registered with `IdentifierWhitelist`. This is important to ensure that the UMA DVM can resolve any disputes for these synthetic tokens. 
+```
+const identifierWhitelist = await IdentifierWhitelist.deployed()
+
```
Comment (chrismaree) [inline documentation/tutorials/creating_synthetic_tokens.md]: To be consistent let's remove the blank line
```suggestion
```

File: documentation/tutorials/creating_synthetic_tokens.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+```
+const registry = await Registry.deployed()
+await registry.addMember(1, empCreator.address)
+```
+ 
+ 
+7. Now, we can create a new expiring multiparty synthetic token with the factory instance.
+```
+const txResult = await empCreator.createExpiringMultiParty(constructorParams)
+
```
Comment (chrismaree) [inline documentation/tutorials/creating_synthetic_tokens.md]: Same as above
```suggestion
```

**Comment by hlambur** [comment]
Thanks @chrismaree! Fwiw Regina and I put this on GitHub partially just as an experiment to see how that would go. This is still in draft form. We're trying to decide if we should write the first versions of the various bits of documentation in google docs or directly in markdown.

**Comment by chrismaree** [comment]
> Thanks @chrismaree! Fwiw Regina and I put this on GitHub partially just as an experiment to see how that would go. This is still in draft form. We're trying to decide if we should write the first versions of the various bits of documentation in google docs or directly in markdown.

ok great. Google docs works well but directly in markdown means you dont have to move across formatting which makes life easier. You can also use vs code or something like [MacDown](https://macdown.uranusjr.com/) for easy markdown formatting and editing on mac.

File: documentation/tutorials/creating_synthetic_tokens.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+ 
+## Step 1: Deploy a contract and create tokens. 
+ 
+1. Set up the truffle environment and migrate contracts. (This step is not necessary if working with an existing testnet or mainnet deployment.)
+```
+$(npm bin)/truffle migrate --network test
+```
+ 
+2. Open the truffle console.
+```
```
Comment (chrismaree) [inline documentation/tutorials/creating_synthetic_tokens.md]: This needs the enviroment like this:
```suggestion
```bash
```

File: documentation/tutorials/creating_synthetic_tokens.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+$(npm bin)/truffle console --network test
+```
+ 
+3. Create an instance of the expiring multiparty creator (the contract factory for synthetic tokens). This command should return “undefined”. 
+```
+const empCreator = await ExpiringMultiPartyCreator.deployed()
+```
+ 
+4. Define the parameters for the synthetic tokens you would like to create. For documentation surrounding the meanings of each parameter, please see this page. 
+
```
Comment (chrismaree) [inline documentation/tutorials/creating_synthetic_tokens.md]: no white space here:
```suggestion
```

File: documentation/tutorials/creating_synthetic_tokens.md [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+ 
+7. Now, we can create a new expiring multiparty synthetic token with the factory instance.
+```
+const txResult = await empCreator.createExpiringMultiParty(constructorParams)
+const emp = await ExpiringMultiParty.at(txResult.logs[0].args.expiringMultiPartyAddress)
+```
+ 
+## How to conduct initial token creation
+ 
+1. Now that we’ve parameterized and deployed the synthetic token contract, we will create synthetic tokens from that contract. The first step is to create an instance of the Test token and mint 10,000 to the wallet. This is the token that will serve as collateral for the synthetic token. Give permission to the empCreator to spend the collateral tokens on our behalf.
```
Comment (hlambur) [inline documentation/tutorials/creating_synthetic_tokens.md]:
```suggestion
1. Now that we’ve parameterized and deployed the synthetic token contract, we will create synthetic tokens from that contract. The first step is to create an instance of the Test token and mint 10,000 to the wallet. This is the token that will serve as collateral for the synthetic token. [Give permission](#how-to-conduct-initial-token-creation) to the empCreator to spend the collateral tokens on our behalf. 
```

File: documentation/tutorials/creating_synthetic_tokens.md [hunk 8]
```diff
@@ -0,0 +1,144 @@
+# Creating Synthetic Tokens from Truffle console
+
+This tutorial will show you how to create synthetic tokens from the command line using UMA’s synthetic token template. Before beginning this tutorial, please make sure your environment is set up correctly by following the instructions in the “Prerequisites” section. After completing this section, your environment will: 
+* Clone the UMA repo
+* Run Ganache on localhost port 9545
```
Comment (nicholaspai) [inline documentation/tutorials/creating_synthetic_tokens.md]: Maybe rephrase to "Run a personal Ethereum blockchain locally on port 9545"

File: documentation/tutorials/creating_synthetic_tokens.md [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+$(npm bin)/truffle migrate --network test
+```
+ 
+2. Open the truffle console.
+```
+$(npm bin)/truffle console --network test
+```
+ 
+3. Create an instance of the expiring multiparty creator (the contract factory for synthetic tokens). This command should return “undefined”. 
+```
```
Comment (nicholaspai) [inline documentation/tutorials/creating_synthetic_tokens.md]: Similar to Chris' comment above, change this to "```js" for better readability

File: documentation/tutorials/creating_synthetic_tokens.md [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+const empCreator = await ExpiringMultiPartyCreator.deployed()
+```
+ 
+4. Define the parameters for the synthetic tokens you would like to create. For documentation surrounding the meanings of each parameter, please see this page. 
+ 
+```
+const constructorParams = { expirationTimestamp: "123456789000", withdrawalLiveness: "1000", siphonDelay: "100000", collateralAddress: TestnetERC20.address, tokenFactoryAddress: TokenFactory.address, priceFeedIdentifier: web3.utils.utf8ToHex("UMATEST"), syntheticName: "Test UMA Token", syntheticSymbol: "UMATEST", liquidationLiveness: "1000", collateralRequirement: { rawValue: web3.utils.toWei("1.5") }, disputeBondPct: { rawValue: web3.utils.toWei("0.1") }, sponsorDisputeRewardPct: { rawValue: web3.utils.toWei("0.1") }, disputerDisputeRewardPct: { rawValue: web3.utils.toWei("0.1") } }
```
Comment (nicholaspai) [inline documentation/tutorials/creating_synthetic_tokens.md]: We no longer need to pass in `siphonDelay`, `tokenFactoryAddress`, `liquidationLiveness`, `withdrawalLiveness` and the `expirationTimestamp` must be one of these:

```
        1585699200, // 2020-04-01T00:00:00.000Z
        1588291200, // 2020-05-01T00:00:00.000Z
        1590969600, // 2020-06-01T00:00:00.000Z
        1593561600, // 2020-07-01T00:00:00.000Z
        1596240000, // 2020-08-01T00:00:00.000Z
        1598918400, // 2020-09-01T00:00:00.000Z
        1601510400, // 2020-10-01T00:00:00.000Z
        1604188800, // 2020-11-01T00:00:00.000Z
        1606780800, // 2020-12-01T00:00:00.000Z
        1609459200, // 2021-01-01T00:00:00.000Z
        1612137600, // 2021-02-01T00:00:00.000Z
        1614556800, // 2021-03-01T00:00:00.000Z
        1617235200, // 2021-04-01T00:00:00.000Z
        1619827200, // 2021-05-01T00:00:00.000Z
        1622505600, // 2021-06-01T00:00:00.000Z
        1625097600, // 2021-07-01T00:00:00.000Z
        1597017600 // 2020-08-10T00:00:00.000Z (Note: This is a special expiration that marks the end of the Tokyo Olympics 2020)
```

File: documentation/tutorials/creating_synthetic_tokens.md [hunk 11]
```diff
@@ -0,0 +1,144 @@
+# Creating Synthetic Tokens from Truffle console
+
+This tutorial will show you how to create synthetic tokens from the command line using UMA’s synthetic token template. Before beginning this tutorial, please make sure your environment is set up correctly by following the instructions in the “Prerequisites” section. After completing this section, your environment will: 
+* Clone the UMA repo
+* Run Ganache on localhost port 9545
+* Compile all contracts in the UMA repo
+ 
+Below, we’ll discuss how to create and manage a token sponsor position via the command line on a local testnet. Please look here for information on existing UMA-supported testnet and mainnet deployments. 
+ 
+## Step 1: Deploy a contract and create tokens.
```
Comment (hlambur) [inline documentation/tutorials/creating_synthetic_tokens.md]: RC, can we change the titles to either all read "step 1, 2, 3" or none say this? (I think I prefer none). Can you also make sure the titles have consistent formatting (All Caps, or just first), and that the titles all flow together cohesively.

File: documentation/tutorials/creating_synthetic_tokens.md (hunk 10 above)
Comment (chrismaree) [reply documentation/tutorials/creating_synthetic_tokens.md]: Here is a suggested param object that fits these new restrictions: 
```
const constructorParams = { expirationTimestamp: "1585699200", collateralAddress: TestnetERC20.address, priceFeedIdentifier: web3.utils.utf8ToHex("UMATEST"), syntheticName: "Test UMA Token", syntheticSymbol: "UMATEST", collateralRequirement: { rawValue: web3.utils.toWei("1.5") }, disputeBondPct: { rawValue: web3.utils.toWei("0.1") }, sponsorDisputeRewardPct: { rawValue: web3.utils.toWei("0.1") }, disputerDisputeRewardPct: { rawValue: web3.utils.toWei("0.1") } }

```

I tested this locally as well tn double check :)

**Comment by chrismaree** [comment]
In addition to the new constructor param we now need to register the collateral within the `collateralTokenWhitelist`. To do this you will need to run two additional commands before calling the `empCreator.createExpiringMultiParty()` function as follows:
```
collateralTokenWhitelist = await AddressWhitelist.at(await empCreator.collateralTokenWhitelist())
await collateralTokenWhitelist.addToWhitelist(TestnetERC20.address)
```

File: documentation/tutorials/creating_synthetic_tokens.md (hunk 8 above)
Comment (rcai1) [reply documentation/tutorials/creating_synthetic_tokens.md]: Good idea - will make this change.

File: documentation/tutorials/creating_synthetic_tokens.md (hunk 11 above)
Comment (rcai1) [reply documentation/tutorials/creating_synthetic_tokens.md]: Yes - I made all these changes in the google doc but have been slower to make them in github.

File: documentation/tutorials/creating_synthetic_tokens.md [hunk 12]
```diff
@@ -1,3 +1,5 @@
+# Creating Synthetic Tokens from Truffle console
+
 This tutorial will show you how to create synthetic tokens from the command line using UMA’s synthetic token template. Before beginning this tutorial, please make sure your environment is set up correctly by following the instructions in the “Prerequisites” section. After completing this section, your environment will: 
 * Clone the UMA repo
 * Run Ganache on localhost port 9545
```
Comment (rcai1) [inline documentation/tutorials/creating_synthetic_tokens.md]: * Run a personal Ethereum blockchain locally on port 9545

File: documentation/tutorials/creating_synthetic_tokens_from_truffle_console.md [hunk 13]
```diff
@@ -0,0 +1,147 @@
...
+ 
+1. Now that we’ve parameterized and deployed the synthetic token contract, we will create synthetic tokens from that contract. The first step is to create an instance of the Test token and mint 10,000 to the wallet. This is the token that will serve as collateral for the synthetic token. Give permission to the empCreator to spend the collateral tokens on our behalf. 
+```js
+const collateralToken = await TestnetERC20.deployed()
+await collateralToken.allocateTo(accounts[0], web3.utils.toWei("10000"))
+await collateralToken.approve(emp.address, web3.utils.toWei("10000"))
+```
+ 
+2. We can now create a synthetic token position. We will deposit 150 units of collateral (the first argument) to create 100 units of synthetic tokens (the second argument).
+
```
Comment (chrismaree) [inline documentation/tutorials/creating_synthetic_tokens_from_truffle_console.md]:
```suggestion
```

File: documentation/tutorials/creating_synthetic_tokens_from_truffle_console.md [hunk 14]
```diff
@@ -0,0 +1,147 @@
...
+```
+ 
+2. We can now create a synthetic token position. We will deposit 150 units of collateral (the first argument) to create 100 units of synthetic tokens (the second argument).
+ 
+```js
+await emp.create({rawValue: web3.utils.toWei("150")},{rawValue:web3.utils.toWei("100")})
+```
+ 
+3. Let’s check that we now have synthetic tokens. We should have 100 synthetic tokens and 9,850 collateral tokens. 
+
```
Comment (chrismaree) [inline documentation/tutorials/creating_synthetic_tokens_from_truffle_console.md]:
```suggestion
```

File: documentation/tutorials/creating_synthetic_tokens_from_truffle_console.md [hunk 15]
```diff
@@ -0,0 +1,147 @@
...
+steps:
+ 
+1. advance timestamp
+2. emp.expire
+3. DVM vote on price
+4. advance time again
+5. emp.settleExpired
+6. check balance
+
+-->
```
Comment (mrice32) [inline documentation/tutorials/creating_synthetic_tokens_from_truffle_console.md]: I think it's fine to leave short descriptive TODOs, but personally, I'm not a huge fan of having a large notes section sitting in the repo. Generally, for stuff like this, we would normally make a github issue that tracks progress on this document and put notes like this in the github issue. Then, you can reference that github issue with a TODO. Something like:
```
TODO: certain sections of this doc are incomplete. See issue #12345 for tracking. 
```
or
```
TODO(#12345): certain sections of this doc are incomplete.
```

Part of the reasoning here is that you don't want to send out a new PR every time you want to update your notes on this doc, since those may change very frequently and aren't really worthy of a review. 

This is a personal preference on my part, and if others disagree, I'm fine with leaving this as is.

File: documentation/tutorials/creating_synthetic_tokens_from_truffle_console.md [hunk 16]
```diff
@@ -0,0 +1,147 @@
+# Creating Synthetic Tokens from Truffle Console
+
+This tutorial will show you how to create synthetic tokens from the command line using UMA’s synthetic token template. Before beginning this tutorial, please make sure your environment is set up correctly by following the instructions in the “Prerequisites” section. After completing this section, your environment will:
```
[the remark this answers is not in the document]
Comment (rcai1) [reply documentation/tutorials/creating_synthetic_tokens_from_truffle_console.md]: 1. good point! Let me add that there. 
2. Sounds good - I trust the language you've selected over mine.

File: documentation/tutorials/creating_synthetic_tokens_from_truffle_console.md [hunk 17]
```diff
@@ -0,0 +1,147 @@
+# Creating Synthetic Tokens from Truffle Console
+
+This tutorial will show you how to create synthetic tokens from the command line using UMA’s synthetic token template. Before beginning this tutorial, please make sure your environment is set up correctly by following the instructions in the “Prerequisites” section. After completing this section, your environment will: 
+* Clone the UMA repo
+* Run a personal Ethereum blockchain locally on port 9545
+* Compile all contracts in the UMA repo
+
+Below, we’ll discuss how to create and manage a token sponsor position via the command line on a local testnet. UMA also has a command-line interface (CLI) tool that bundles and renames some of these steps. These tutorials discuss how to create and manage a token sponsor position using this CLI tool. <!-- (links to separate page) -->
```
[the remark this answers is not in the document]
Comment (rcai1) [reply documentation/tutorials/creating_synthetic_tokens_from_truffle_console.md]: noted, will fix.

File: documentation/tutorials/creating_synthetic_tokens_from_truffle_console.md [hunk 18]
```diff
@@ -0,0 +1,147 @@
...
+ 
+## Deposit and withdraw collateral
+ 
+1. As a token sponsor, we may wish to add additional collateral to our position to avoid being liquidated by a token holder. Let’s deposit 10 additional collateral tokens to our position and see our updated balance, from 9,925 to 9,915. 
+```js
+await emp.deposit({rawValue: web3.utils.toWei("10")})
+(await collateralToken.balanceOf(accounts[0])).toString()
+```
+ 
+2. For a token sponsor to withdraw collateral from his position, there are typically 2 ways to do this. Click here for more information. In this scenario, because we are the only token sponsor, we will have to withdraw collateral the “slow” way. First, we need to request a withdrawal of 10 collateral tokens.
```
[the remark this answers is not in the document]
Comment (rcai1) [reply documentation/tutorials/creating_synthetic_tokens_from_truffle_console.md]: indeed, we just did not add a link yet.

File: documentation/tutorials/creating_synthetic_tokens_from_truffle_console.md (hunk 15 above)
Comment (rcai1) [reply documentation/tutorials/creating_synthetic_tokens_from_truffle_console.md]: Makes sense - will change.

File: documentation/tutorials/creating_synthetic_tokens_from_truffle_console.md [hunk 19]
```diff
@@ -0,0 +1,151 @@
...
+* Have the protocol repo cloned.
+* Be running an instance of Ganache on port 9545.
+* Have run truffle migrations for the contracts in `core/`.
+
+Below, we’ll discuss how to create and manage a token sponsor position via the command line on a local testnet. 
+UMA also has a command-line interface (CLI) tool that bundles and renames some of these steps. 
+These tutorials discuss how to create and manage a token sponsor position using this CLI tool. 
+<!-- TODO: links to separate page -->
+
+* Begin [here](#parameterize-and-deploy-a-contract) if you are creating a new type of synthetic token on a local testnet.
```
Comment (mrice32) [inline documentation/tutorials/creating_synthetic_tokens_from_truffle_console.md]: @rcai1 I know we've discussed this breakage, but could you make an issue under the new documentation milestone I created? https://github.com/UMAprotocol/protocol/milestone/12

Comment (rcai1) [reply documentation/tutorials/creating_synthetic_tokens_from_truffle_console.md]: Done - please see https://github.com/UMAprotocol/protocol/issues/1055

File: documentation/tutorials/creating_synthetic_tokens_from_truffle_console.md [hunk 20]
```diff
@@ -0,0 +1,151 @@
...
+(await collateralToken.balanceOf(accounts[0])).toString()
+// collateral token balance
+```
+<!-- 
+
+--END OF TUTORIAL--
+Notes: We might prefer to show people how to settle a contract after expiration using a CLI tool so they can change between token sponsor, token holder, and DVM voter personas more easily. 
+
+This is particularly relevant for settling a token after the expiration time, when as a token sponsor they might settle, as a voter they might decide on the final token redemption value, and as token holder they might redeem. 
+-->
```
Comment (mrice32) [inline documentation/tutorials/creating_synthetic_tokens_from_truffle_console.md]: Still not a huge fan of this notes section, but since it's smaller and more polished, I think that's fine.
